akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -917,24 +919,33 @@ abstract class ShardCoordinator(
         sender() ! reply
 
       case ShardCoordinator.Internal.Terminate =>
-        if (rebalanceInProgress.isEmpty)
-          log.debug("{}: Received termination message.", typeName)
-        else if (log.isDebugEnabled) {
-          if (verboseDebug)
-            log.debug(
-              "{}: Received termination message. Rebalance in progress of [{}] shards [{}].",
-              typeName,
-              rebalanceInProgress.size,
-              rebalanceInProgress.keySet.mkString(", "))
-          else
-            log.debug(
-              "{}: Received termination message. Rebalance in progress of [{}] shards.",
-              typeName,
-              rebalanceInProgress.size)
-        }
-        context.stop(self)
+        terminate()
     }: Receive).orElse[Any, Unit](receiveTerminated)
 
+  private def terminate(): Unit = {
+    if (localRegionTerminationInProgress) {
```

Comment:
I don't think we need an extra `var` for this. Look for local address path in `gracefulShutdownInProgress`?

Reply:
It's about having observed `GracefulShutdownReq` from the local region and then that termination completing vs just knowing that the coordinator when we get  `Terminate`, but maybe you are right.

Reply:
Isn't the flow like this:
1. receive GracefulShutdownReq, adding the ref to gracefulShutdownInProgress
2. receive Terminate, at this point the gracefulShutdownInProgress still contains the local regions ref
3. regionTerminated

Reply:
Ah, yes, you are right. 👍